test/e2e/sgx/sgx.go

```diff
@@ -48,104 +48,116 @@ func describe() {
+				framework.ExpectNoError(err, "pod Create API error")
+
+				ginkgo.By("waiting the pod to finish successfully")
+				e2epod.NewPodClient(f).WaitForSuccess(ctx, pod.ObjectMeta.Name, 60*time.Second)
+			})
+		})
 	})
 
-	ginkgo.It("deploys SGX plugin with operator", func(ctx context.Context) {
-		utils.Kubectl("", "apply", "-k", "deployments/operator/default/kustomization.yaml")
+	ginkgo.Describe("With using operator", func() {
+		ginkgo.It("deploys SGX plugin with operator", func(ctx context.Context) {
+			utils.Kubectl("", "apply", "-k", "deployments/operator/default/kustomization.yaml")
 
-		if _, err := e2epod.WaitForPodsWithLabelRunningReady(ctx, f.ClientSet, ns, labels.Set{"control-plane": "controller-manager"}.AsSelector(), 1, timeout); err != nil {
-			framework.Failf("unable to wait for all pods to be running and ready: %v", err)
-		}
+			if _, err := e2epod.WaitForPodsWithLabelRunningReady(ctx, f.ClientSet, ns, labels.Set{"control-plane": "controller-manager"}.AsSelector(), 1, timeout); err != nil {
+				framework.Failf("unable to wait for all pods to be running and ready: %v", err)
+			}
 
-		utils.Kubectl("", "apply", "-f", "deployments/operator/samples/deviceplugin_v1_sgxdeviceplugin.yaml")
+			utils.Kubectl("", "apply", "-f", "deployments/operator/samples/deviceplugin_v1_sgxdeviceplugin.yaml")
 
-		if _, err := e2epod.WaitForPodsWithLabelRunningReady(ctx, f.ClientSet, ns, labels.Set{"app": "intel-sgx-plugin"}.AsSelector(), 1, timeout); err != nil {
-			framework.Failf("unable to wait for all pods to be running and ready: %v", err)
-		}
+			if _, err := e2epod.WaitForPodsWithLabelRunningReady(ctx, f.ClientSet, ns, labels.Set{"app": "intel-sgx-plugin"}.AsSelector(), 1, timeout); err != nil {
+				framework.Failf("unable to wait for all pods to be running and ready: %v", err)
+			}
 
-		if err := utils.WaitForNodesWithResource(ctx, f.ClientSet, "sgx.intel.com/epc", 150*time.Second); err != nil {
-			framework.Failf("unable to wait for nodes to have positive allocatable epc resource: %v", err)
-		}
+			waitForNodesWithSgxResources(ctx, f)
 
-		if err := utils.WaitForNodesWithResource(ctx, f.ClientSet, "sgx.intel.com/enclave", 30*time.Second); err != nil {
-			framework.Failf("unable to wait for nodes to have positive allocatable enclave resource: %v", err)
-		}
+			utils.Kubectl("", "delete", "-f", "deployments/operator/samples/deviceplugin_v1_sgxdeviceplugin.yaml")
 
-		if err := utils.WaitForNodesWithResource(ctx, f.ClientSet, "sgx.intel.com/provision", 30*time.Second); err != nil {
-			framework.Failf("unable to wait for nodes to have positive allocatable provision resource: %v", err)
-		}
+			utils.Kubectl("", "delete", "-k", "deployments/operator/default/kustomization.yaml")
+		})
+	})
+}
 
-		utils.Kubectl("", "delete", "-f", "deployments/operator/samples/deviceplugin_v1_sgxdeviceplugin.yaml")
+func waitForNodesWithSgxResources(ctx context.Context, f *framework.Framework) {
```

Comment:
we could think about can we check this in one place only for both operator and non-operator testing (now the operator rework PR moves the check to utils)